src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -337,25 +342,46 @@
                 });
     }
 
+    public Uni<List<Endpoint>> loadProperties(List<Endpoint> endpoints) {
+        // Group endpoints in types and load in batches each type.
+        // Currently only webhook types have properties.
+        endpoints = endpoints.stream().distinct().collect(Collectors.toList());
+
+        Map<UUID, Endpoint> webhookEndpoints = endpoints
+                .stream()
+                .filter(endpoint -> endpoint.getType().equals(EndpointType.WEBHOOK))
+                .collect(Collectors.toMap(Endpoint::getId, Function.identity()));
+
+        final Uni<List<Endpoint>> webhookUnis;
+
+        if (webhookEndpoints.size() > 0) {
+            webhookUnis = session
+                    .find(EndpointWebhook.class, webhookEndpoints.keySet().toArray())
+                    .onItem().transform(endpointWebhooks -> endpointWebhooks
+                            .stream()
+                            .map(webhook -> {
+                                Endpoint endpoint = webhookEndpoints.get(webhook.getId());
+                                WebhookAttributes properties = buildWebhookAttributes(webhook);
+                                endpoint.setProperties(properties);
+                                return endpoint;
+                            })
+                            .collect(Collectors.toList())
+                    );
+        } else {
+            webhookUnis = Uni.createFrom().item(Collections.emptyList());
+        }
+
+        return webhookUnis.replaceWith(endpoints);
```

Comment:
It should be possible to get rid of that returned value and return `Uni<Void>` instead. There's only one case where this is not possible but it could be transformed to an `onItem().call()` execution:

https://github.com/RedHatInsights/notifications-backend/blob/3ac94434be11f81ebd3b2ec2916c17c60361e2a6/src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java#L172

Reply:
You mean this whole function to return Uni<Void>?

Reply:
Yes, because Mutiny discards the returned value during an `onItem().call()` execution.

Reply:
It only needs to know when the execution is done before proceeding downstream.